test/extended/kubevirt/nodereboot.go

```diff
@@ -0,0 +1,85 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("[sig-kubevirt] node reboot", func() {
+	oc := exutil.NewCLIWithPodSecurityLevel("ns-global", admissionapi.LevelBaseline)
+
+	InKubeVirtClusterContext(oc, func() {
+		mgmtFramework := e2e.NewDefaultFramework("mgmt-framework")
+		mgmtFramework.SkipNamespaceCreation = true
+
+		hostedFramework := e2e.NewDefaultFramework("hosted-framework")
+		hostedFramework.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+		When("an hosted control plane worker is rebooted", func() {
+			var hostedClusterNodeToBeRebooted corev1.Node
+
+			isNodeReady := func(nodeName string) (bool, error) {
+				node, err := hostedFramework.ClientSet.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+				if k8sapierrors.IsNotFound(err) {
+					return false, err
+				}
+				for _, condition := range node.Status.Conditions {
+					if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+				return false, nil
+			}
+
+			BeforeEach(func() {
+				nodeList, err := hostedFramework.ClientSet.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(nodeList.Items).NotTo(BeEmpty())
+
+				hostedClusterNodeToBeRebooted = nodeList.Items[0]
+
+				Eventually(func() (bool, error) { return isNodeReady(hostedClusterNodeToBeRebooted.Name) }).
+					WithTimeout(2*time.Minute).
+					WithPolling(5*time.Second).
+					Should(BeTrue(), fmt.Sprintf("node %q should have the `Ready` condition", hostedClusterNodeToBeRebooted.Name))
+
+				setMgmtFramework(mgmtFramework)
+				expectNoError(
+					rebootWorkerNode(
+						mgmtFramework,
+						hostedClusterNodeToBeRebooted.Name,
+					),
+				)
+			})
+
+			It("should maintain node readiness", func() {
+				Eventually(func() (bool, error) { return isNodeReady(hostedClusterNodeToBeRebooted.Name) }).
+					WithTimeout(2*time.Minute).
+					WithPolling(5*time.Second).
+					Should(
+						BeTrue(),
+						fmt.Sprintf("node %q should have the `Ready` condition", hostedClusterNodeToBeRebooted.Name),
+					)
```

Comment:
We can have a false possitive here, since it takes some time for the node to go NotReady. 

So we should first wait until node is NotReady, then Check that goes to Ready and then check that it stay that way.

Reply:
Done.